lib/jekyll/post.rb

```diff
@@ -280,6 +280,7 @@ def destination(dest)
       # The url needs to be unescaped in order to preserve the correct filename
       path = site.in_dest_dir(dest, URL.unescape_path(url))
       path = File.join(path, "index.html") if self.url =~ /\/$/
```

Comment:
also this should totally be `.end_with?("/")`... not sure if that's for this PR but it's an optimization we should take advantage of. i'll write a benchmark to ensure this